Common/Orders/MarketOrder.cs

```diff
@@ -42,6 +42,49 @@ public override decimal Value
         }
 
         /// <summary>
+        /// Create update request for pending orders. Null values will be ignored.
+        /// </summary>
+        public UpdateOrderRequest CreateUpdateRequest(int? quantity = null, string tag = null)
+        {
+            return new UpdateOrderRequest
+            {
+                Id = Guid.NewGuid(),
+                OrderId = Id,
+                Created = DateTime.Now,
+                Quantity = quantity ?? Quantity,
+                Tag = tag ?? Tag
+            };
+        }
+
+        /// <summary>
+        /// Create submit request.
+        /// </summary>
+        public static SubmitOrderRequest CreateSubmitRequest(SecurityType securityType, string symbol, int quantity, DateTime time, decimal price = 0, string tag = null)
+        {
```

Comment:
Price parameter here is the *fill* price, so it can't be updated manually by the user or anyone.